File: dotnet/src/canvas/mxStencilRegistry.cs
using System;
using System.Collections.Generic;

namespace com.mxgraph
{
    public class mxStencilRegistry
    {

        protected static Dictionary<string, mxStencil> stencils = new Dictionary<string, mxStencil>();

        /// <summary>
        /// Adds the given stencil.
        /// </summary>
        /// <param name="name"></param>
        /// <param name="stencil"></param>
        public static void AddStencil(string name, mxStencil stencil)
        {
            stencils[name] = stencil;
        }

        /// <summary>
        /// Returns the stencil for the given name.
        /// </summary>
        /// <param name="name"></param>
        /// <returns></returns>
        public static mxStencil GetStencil(string name)
        {
            if (stencils.ContainsKey(name))
            {
                return stencils[name];
            }

            return null;
        }

    }
}

File: javascript/src/js/shape/mxStencilRegistry.js
/**
 * $Id: mxStencilRegistry.js,v 1.2 2011-07-15 12:57:50 gaudenz Exp $
 * Copyright (c) 2006-2010, JGraph Ltd
 * 
 * Code to add stencils.
 * 
 * (code)
 * var req = mxUtils.load('test/stencils.xml');
 * var root = req.getDocumentElement();
 * var shape = root.firstChild;
 * 
 * while (shape != null)
 * {
 * 	 if (shape.nodeType == mxConstants.NODETYPE_ELEMENT)
 *   {
 *     mxStencilRegistry.addStencil(shape.getAttribute('name'), new mxStencil(shape));
 *   }
 *   
 *   shape = shape.nextSibling;
 * }
 * (end)
 */
var mxStencilRegistry =
{
	/**
	 * Class: mxStencilRegistry
	 * 
	 * A singleton class that provides a registry for stencils and the methods
	 * for painting those stencils onto a canvas or into a DOM.
	 */
	stencils: [],
	
	/**
	 * Function: addStencil
	 * 
	 * Adds the given <mxStencil>.
	 */
	addStencil: function(name, stencil)
	{
		mxStencilRegistry.stencils[name] = stencil;
	},
	
	/**
	 * Function: getStencil
	 * 
	 * Returns the <mxStencil> for the given name.
	 */
	getStencil: function(name)
	{
		return mxStencilRegistry.stencils[name];
	}

};
